File: backend/src/Connect.Application/Common/Diagnostics/DiagnosticEvent.cs
namespace Connect.Application.Common.Diagnostics;

public class DiagnosticEvent
{
    public string Id { get; set; } = string.Empty;
    public string Timestamp { get; set; } = string.Empty;
    public string Severity { get; set; } = "Info";
    public string Component { get; set; } = string.Empty;
    public string EventName { get; set; } = string.Empty;
    public string Message { get; set; } = string.Empty;
    public string? UserId { get; set; }
    public string? SessionId { get; set; }
    public string? CorrelationId { get; set; }
    public string? CallId { get; set; }
    public object? Metadata { get; set; }
}

File: frontend/lib/models/diagnostic_event.dart
class DiagnosticEvent {
  final String id;
  final String timestamp;
  final String severity;
  final String component;
  final String eventName;
  final String message;
  final String? userId;
  final String? sessionId;
  final String? correlationId;
  final String? callId;
  final Map<String, dynamic>? metadata;

  DiagnosticEvent({
    required this.id,
    required this.timestamp,
    required this.severity,
    required this.component,
    required this.eventName,
    required this.message,
    this.userId,
    this.sessionId,
    this.correlationId,
    this.callId,
    this.metadata,
  });

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'timestamp': timestamp,
      'severity': severity,
      'component': component,
      'eventName': eventName,
      'message': message,
      if (userId != null) 'userId': userId,
      if (sessionId != null) 'sessionId': sessionId,
      if (correlationId != null) 'correlationId': correlationId,
      if (callId != null) 'callId': callId,
      if (metadata != null) 'metadata': metadata,
    };
  }

  factory DiagnosticEvent.fromJson(Map<String, dynamic> json) {
    return DiagnosticEvent(
      id: json['id'],
      timestamp: json['timestamp'],
      severity: json['severity'],
      component: json['component'],
      eventName: json['eventName'],
      message: json['message'],
      userId: json['userId'],
      sessionId: json['sessionId'],
      correlationId: json['correlationId'],
      callId: json['callId'],
      metadata: json['metadata'],
    );
  }
}
